x-pack/legacy/plugins/siem/public/components/ml/conditional_links/remove_kql_variables.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RisonValue, encode } from 'rison-node';
+import { decodeRison, isRisonObject, isRegularString } from './rison_helpers';
+
+export const operators = ['and', 'or', 'not'];
+
+export const removeKqlVariablesUsingRegex = (expression: string) => {
+  const myRegexp = /(\s+)*(and|or|not){0,1}(\s+)*([\w\.\-\[\]]+)\s*:\s*"(\$[\w\.\-\(\)\[\]]+\$)"(\s+)*(and|or|not){0,1}(\s+)*/g;
+  return expression.replace(myRegexp, replacement);
+  // return expression.replace(myRegexp, '');
+};
+
+export const replacement = (match: string, ...parts: string[]): string => {
+  if (parts == null) {
+    return '';
+  }
+  const operatorsMatched = parts.reduce<string[]>((accum, part) => {
+    if (part != null && operators.includes(part)) {
+      accum = [...accum, part];
+      return accum;
+    } else {
+      return accum;
+    }
```

Comment:
This `reduce`'s conditional can be condensed ala:

``` ts
return [...accum, ...(part != null && operators.includes(part) ? [part] : [])];
```

🥔, 🍅 though from a readability standpoint -- just something I've been seeing for these filter + reduce situations

Reply:
There's a `if (operatorsMatched.length > 1) {` below this that it has to touch so that return would return early all cases. But we can zoom after we get a breather and see if maybe we can make these look better.

I am happy there are unit tests so refactoring will feel "safe" for us.